### PR TITLE
Fix recall again

### PR DESCRIPTION
**Pull request by Hdt80bro**
Adds better recall reporting in the log and handles defeated brains better.

Partially tested for now.

### CODE REVIEW

File: lua/sim/Recall.lua [hunk 1]
```diff
@@ -185,19 +163,29 @@ local function RecallVotingThread(requestingArmy)
...
+                if brain.RecallVote then
+                    yesVotes = yesVotes + 1
+                else
+                    noVotes = noVotes + 1
+                end
             end
-            brain.RecallVote = nil
             brain.LastRecallVoteTime = gametick
         end
+        brain.RecallVote = nil -- make sure defeated players get reset too
```
Comment (lL1l1) [inline lua/sim/Recall.lua:186]: So this is the critical line that fixes recall frequently failing? Previously the `RecallVote` wasn't cleared when a brain was defeated, so their vote from previous recalls remained locked in for the rest of the game?

The rest is refactoring for better logging and improving the UI behavior to work with 3 player votes when the third is defeated and it transforms into 2 players?

Comment (Hdt80bro) [reply lua/sim/Recall.lua:186]: If you take the conditions of the if statement into account, I guess, yeah.

**Comment by lL1l1** [comment]
([Discord Thread](https://discord.com/channels/197033481883222026/1292616896964853863))
> You said it's partially tested, what needs to be tested?

> Ideally, most of the interactions - but specifically, a team of 4 with one player being defeated before / during the vote, and probably again with a team of 3.
Oh, and there needs to have been a prior failed vote where the defeated player voted no.

## Testing of recall:
Setup is 4v4 launched via the new script.


1. Recall -> deny: recall failed
2. Recall -> accept -> deny: recall failed
  - 1 and 2 are a Bug because we have 4 players and 3 are needed. The code should wait for the 2nd denying vote.
 3. Recall -> accept -> accept: recall succeeded
 4. Recall -> accept -> deny -> recall fail -> denier ctrl-k -> recall (tested that dead player can't recall) ->
   1. Recall -> accept -> recall failed (2-0, 1 abstained)
   2. Recall -> deny -> recall failed
 5. 3 players -> recall -> player who started recall ctrl-k -> UI breaks
{AAE3E780-D43D-45DD-8E9F-5A004DB4D56B}

## Other Notes: 
- It is confusing that the UI keeps the old votes visible, even though the logic always resets players to an abstaining state every vote. It also seems to not maintain the per-slot player vote regardless.
- Sometimes the voting panel can stay extended for a while, not sure why.

**Comment by lL1l1** [comment]
## Test iteration 2:
1. Recall -> deny: recall continues
2. Recall -> accept -> deny: recall continues
 3. Recall -> accept -> accept: recall succeeded
 4. Recall -> accept -> deny -> recall fail -> denier ctrl-k -> recall (tested that dead player can't recall) ->
   1. Recall -> accept -> recall continues
   2. Recall -> deny -> recall failed (this is correct, as 3/3 is needed)
 5. 3 players -> recall -> player who started recall ctrl-k -> UI breaks
 6. 2 players -> recall -> player who did not start vote ctrl-k -> UI breaks too
  - seems like the UI breaks every time it has to update

## Other notes:
- "Not ready for recall" and "recalling" stay on the voting panel in a 2 player team.

**Comment by lL1l1** [comment]
I think all the issues I've highlighted are fixed. Opening 8 instances and voting in different ways along with 1 player dying during the middle of the vote all works as expected sim and UI wise.

**Comment by BlackYps** [comment]
Note to self: When testing, check if the wording of the recall process can be improved to better convey that recalling is surrendering.
One idea: "forfeit the game by recalling from battle?"

**Review by lL1l1 — APPROVED** [review]

**Pull request merged**